catalog/src/main/java/it/eng/catalog/service/CatalogService.java

```diff
@@ -249,4 +236,26 @@ private Catalog.Builder returnBaseCatalogForUpdate(String id) {
                         .createdBy(c.getCreatedBy()))
                 .orElseThrow(() -> new CatalogNotFoundAPIException("Catalog with id: " + id + " not found"));
     }
+
+	public boolean validateOffer(Offer offer) {
+		boolean valid = false;
+		Catalog catalog = getCatalog();
+
+		Offer existingOffer = catalog.getDataset().stream().flatMap(dataset -> dataset.getHasPolicy().stream())
+				.filter(of -> of.getId().equals(offer.getId())).findFirst().orElse(null);
+
+		log.debug("Offer with id '{}' {}", offer.getId(), existingOffer != null ? " found." : "not found.");
+
+		if (existingOffer == null) {
+			valid = false;
```

Comment:
Can you add log line?
log.warn("Offer with id {} not found in catalog", offer.getId());
so we know why offer is not validated?

Reply:
No problemo